### DISCUSSION TITLE
HELP!

### DISCUSSION THREAD

**Discussion by KMH418** [soxoj/socid-extractor#132] [Q&A] [answered]
Can someone PLEASE help me understand this? HOW do I use it? I don't know much about coding. All I've done was on myspace YEARS ago. What do I do with the code, to use it as is?

**Comment by C3n7ral051nt4g3ncy** [comment, marked as answer]
Hi @KMH418, 

Open your terminal and use the following commands:

- `git clone https://github.com/soxoj/socid-extractor`
- then type `cd socid-extractor `
- then `pip3 install -r requirements.txt`
- as a test, use this command: `socid_extractor --url https://www.deviantart.com/muse1908`

Make sure you have Python Installed,[Download Here](https://www.python.org/downloads/)
Pip3 should come with python, after the python download, to check type `pip3 --version `

Git should be on your computer already, type `git --version` to check.
if it's not installed, simply install it | [See tutorial](https://www.jcchouinard.com/install-git/)

**Reply by KMH418** [reply]
Thank you for your reply! 
I am completely new to ALL of this!! I honestly need "children's dictionary" explanations of how to do it! So I did the python and the Git installs. What is my terminal? Is that on githib or python or my computer command?

**Reply by C3n7ral051nt4g3ncy** [reply]
@KMH418

Are you on macOS or Windows ?

**Reply by KMH418** [reply]
windows

**Reply by C3n7ral051nt4g3ncy** [reply]
@KMH418 Open your start menu and type `cmd` or `Command Prompt`, this should show you the terminaL.
See this video: https://www.youtube.com/watch?v=8Iyldhkrh7E
Basically you will run the tools/scripts from the terminal.

**Reply by KMH418** [reply]
This is what i see after inputting all that: 
C:\Users\chari>git clone https://github.com/soxoj/socid-extractor
fatal: destination path 'socid-extractor' already exists and is not an empty directory.

C:\Users\chari>cd socid-extractor

C:\Users\chari\socid-extractor>pip3 install -r requirements.txt
'pip3' is not recognized as an internal or external command,
operable program or batch file.

C:\Users\chari\socid-extractor>socid_extractor --url https://www.deviantart.com/muse1908
'socid_extractor' is not recognized as an internal or external command,
operable program or batch file.

C:\Users\chari\socid-extractor>

**Reply by KMH418** [reply]
I have both the Git and the python installed...

**Reply by C3n7ral051nt4g3ncy** [reply]
@KMH418 You have obviously already cloned it if the path already exists.

**Reply by C3n7ral051nt4g3ncy** [reply]
each command should be typed separately

**Reply by C3n7ral051nt4g3ncy** [reply]
just do `cd socid_extractor `, and press enter then wait.

**Reply by C3n7ral051nt4g3ncy** [reply]
once you have done cd (change directory), then install the requirements.txt file by doing` pip3 install -r requirements.txt` and press enter

**Reply by KMH418** [reply]
C:\Users\chari>cd socid_extractor
The system cannot find the path specified.

**Reply by KMH418** [reply]
Last night after numerous google searches, I cloned the repository into GitHub for the web version and the desktop application version. Then I didn't know where to go from there... 

A lot of the times when I find something I need to use that refers me to a GitHub site, there is a live webpage running and that's what I use. I haven't ever had to set one up myself or run code.

**Reply by C3n7ral051nt4g3ncy** [reply]
> C:\Users\chari>cd socid_extractor
> 
> The system cannot find the path specified.
> 
> 

- You did: cd socid_extractor

- it should be cd socid-extractor

**Reply by KMH418** [reply]
Ok. I did a copy and paste from somewhere. Let me try that.

**Reply by KMH418** [reply]
This is all that did. 

C:\Users\chari>cd socid-extractor

C:\Users\chari\socid-extractor>

**Reply by C3n7ral051nt4g3ncy** [reply]
Great. Now do the next command to install requirements.txt and press enter.
After that the final command to start the script

**Reply by KMH418** [reply]
I did the pip right after that and its still saying the same thing

**Reply by C3n7ral051nt4g3ncy** [reply]
Type this: 

python get-pip.py

**Reply by C3n7ral051nt4g3ncy** [reply]
If pip3 install -r requirements.txt did not work, try:
pip install -r requirements.txt

**Reply by KMH418** [reply]
C:\Users\chari\socid-extractor>python get-pip.py
Python was not found; run without arguments to install from the Microsoft Store, or disable this shortcut from Settings > Manage App Execution Aliases.

C:\Users\chari\socid-extractor>pip install -r requirements.txt
'pip' is not recognized as an internal or external command,
operable program or batch file.

**Reply by C3n7ral051nt4g3ncy** [reply]
@KMH418 it doesn't seem as though you installed Python correctly.

Follow this tutorial: 

https://phoenixnap.com/kb/how-to-install-python-3-windows

**Reply by KMH418** [reply]
Ok so that popped up a screen to modify or uninstall. I did an uninstall to reinstall it from here. (I found something a bit ago that told me to do a modify and click to add pip as a path. and it went through the steps to modify the path in the computer. Once I did that and entered it into command, it worked! Then it stopped and went back to saying that it couldn't be found.) grr.

**Reply by C3n7ral051nt4g3ncy** [reply]
Have you typed:
python3 --version?
This will tell you if you I have python installed.

**Reply by KMH418** [reply]
Python comes back with info. I did the command line from the instructions in the link. Pip still doesn't.

**Reply by C3n7ral051nt4g3ncy** [reply]
Ok great we are nearly there. So you have python installed.

Now you need to install pip.

**Reply by KMH418** [reply]
C:\Users\chari>python
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.

C:\Users\chari>python3 --version
Python was not found; run without arguments to install from the Microsoft Store, or disable this shortcut from Settings > Manage App Execution Aliases.


C:\Users\chari>pip -V
'pip' is not recognized as an internal or external command,
operable program or batch file.

**Reply by C3n7ral051nt4g3ncy** [reply]
type this:
py -m pip --version

**Reply by C3n7ral051nt4g3ncy** [reply]
Then try this:

py -m ensurepip --default-pip

**Reply by C3n7ral051nt4g3ncy** [reply]
And if nothing happens it means pip isn't installed.
So do this command :

python get-pip.py

After that you will be all set up I use the tool.

Do the same as before:

- cd socid-extractor and enter 

- pip install -r requirements.txt and enter

**Reply by C3n7ral051nt4g3ncy** [reply]
Once you have done all that:

Type this:

socid_extractor --url https://www.deviantart.com/muse1908

(Notice that once you have changed directory, to launch the tool you need to write in socid_extractor and not socid-extractor)

If nothing works there really is nothing more I can do at distance and means there is another issue with your computer.

**Reply by KMH418** [reply]
https://phoenixnap.com/kb/install-pip-windows
I followed the instructions that were in the first link you sent me and I got things coming back now.

Its finally showing PIP installed. I will try to run the Socid scripts again.

**Reply by KMH418** [reply]
OK i  cant get it to do anything with the socid_extractor --url https://www.deviantart.com/muse1908 script. 

C:\Users\chari\socid-extractor>socid_extractor --url https://www.deviantart.com/muse1908
'socid_extractor' is not recognized as an internal or external command,
operable program or batch file.

**Reply by C3n7ral051nt4g3ncy** [reply]
Awesome, I gave you commands for Mac at the start so that's probably why you didn't get stuff right away.
By using the 'pip' command instead of ´pip3´ this may help, and same with python instead of python3

It all depends on version.

So this topic can be closed you need to mark question as 'answered'.

**Reply by KMH418** [reply]
But is it not supposed to bring back info with the socid_extractor --url https://www.deviantart.com/muse1908 ?

**Reply by C3n7ral051nt4g3ncy** [reply]
> But is it not supposed to bring back info with the socid_extractor --url https://www.deviantart.com/muse1908 ?

You are correct, it should bring back results, meaning there is something that has not been installed correctly.

**Reply by KMH418** [reply]
So I'm guessing this is an issue with how I downloaded the repository, not the python/pip... Should I go back and DL the zip?

**Reply by C3n7ral051nt4g3ncy** [reply]
There is no need to undo any zip for the repository.
If you installed git correctly, all you need to do is this:

`git clone https://github.com/soxoj/socid-extractor`

If you already have it, delete it and start a fresh, don't forget that once you have changed directory, to install the requirements.txt file.

**Reply by KMH418** [reply]
C:\Users\chari\socid-extractor>pip3 install -r requirements.txt
WARNING: Ignoring invalid distribution -ip (c:\users\chari\appdata\local\programs\python\python310\lib\site-packages)
WARNING: Ignoring invalid distribution -ip (c:\users\chari\appdata\local\programs\python\python310\lib\site-packages)
Requirement already satisfied: requests>=2.24.0 in c:\users\chari\appdata\local\programs\python\python310\lib\site-packages (from -r requirements.txt (line 1)) (2.28.1)
Requirement already satisfied: python-dateutil>=2.8.1 in c:\users\chari\appdata\local\programs\python\python310\lib\site-packages (from -r requirements.txt (line 2)) (2.8.2)
Requirement already satisfied: beautifulsoup4~=4.11.1 in c:\users\chari\appdata\local\programs\python\python310\lib\site-packages (from -r requirements.txt (line 3)) (4.11.1)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in c:\users\chari\appdata\local\programs\python\python310\lib\site-packages (from requests>=2.24.0->-r requirements.txt (line 1)) (1.26.12)
Requirement already satisfied: charset-normalizer<3,>=2 in c:\users\chari\appdata\local\programs\python\python310\lib\site-packages (from requests>=2.24.0->-r requirements.txt (line 1)) (2.1.1)
Requirement already satisfied: certifi>=2017.4.17 in c:\users\chari\appdata\local\programs\python\python310\lib\site-packages (from requests>=2.24.0->-r requirements.txt (line 1)) (2022.6.15)
Requirement already satisfied: idna<4,>=2.5 in c:\users\chari\appdata\local\programs\python\python310\lib\site-packages (from requests>=2.24.0->-r requirements.txt (line 1)) (3.3)
Requirement already satisfied: six>=1.5 in c:\users\chari\appdata\local\programs\python\python310\lib\site-packages (from python-dateutil>=2.8.1->-r requirements.txt (line 2)) (1.16.0)
Requirement already satisfied: soupsieve>1.2 in c:\users\chari\appdata\local\programs\python\python310\lib\site-packages (from beautifulsoup4~=4.11.1->-r requirements.txt (line 3)) (2.3.2.post1)
WARNING: Ignoring invalid distribution -ip (c:\users\chari\appdata\local\programs\python\python310\lib\site-packages)
WARNING: Ignoring invalid distribution -ip (c:\users\chari\appdata\local\programs\python\python310\lib\site-packages)
WARNING: Ignoring invalid distribution -ip (c:\users\chari\appdata\local\programs\python\python310\lib\site-packages)
WARNING: Ignoring invalid distribution -ip (c:\users\chari\appdata\local\programs\python\python310\lib\site-packages)

C:\Users\chari\socid-extractor>socid_extractor --url https://www.deviantart.com/muse1908
'socid_extractor' is not recognized as an internal or external command,
operable program or batch file.

**Reply by KMH418** [reply]
I dont think that the Socid file is going to the correct place on my comp. Its here: C:\Users\chari\socid-extractor

**Comment by C3n7ral051nt4g3ncy** [comment]
https://github.com/soxoj/socid-extractor/discussions/132#discussioncomment-3551514

**Reply by C3n7ral051nt4g3ncy** [reply]
its not possible that you don't have pip3 as it now comes bundles with new versions of python. to see if you have it correctly type:
`pip3 --version` and enter

**Reply by KMH418** [reply]
C:\Users\chari>pip3 --version
'pip3' is not recognized as an internal or external command,
operable program or batch file.

**Reply by C3n7ral051nt4g3ncy** [reply]
See screenshot from my terminal:

<img width="1279" src="https://user-images.githubusercontent.com/104733166/188499612-889fcdec-e672-40a5-92b3-69687993c1c4.png">

**Comment by C3n7ral051nt4g3ncy** [comment]
@KMH418 When you did `cd socid-extractror`, did you actually then do pip install `-r requirements.txt`?

**Reply by KMH418** [reply]
"Open your terminal and use the following commands:

git clone https://github.com/soxoj/socid-extractor
then type cd socid-extractor
then pip3 install -r requirements.txt
as a test, use this command: socid_extractor --url https://www.deviantart.com/muse1908 "

I just did all this ^^ after deleting the pre-existing Socid file from my comp.

**Reply by C3n7ral051nt4g3ncy** [reply]
and?

**Reply by KMH418** [reply]
Everything is good until i give the final command and it tells me the same thing:

C:\Users\chari\socid-extractor>socid_extractor --url https://www.deviantart.com/muse1908
'socid_extractor' is not recognized as an internal or external command,
operable program or batch file.

I've deleted the socid file and reinstalled it. I've uninstalled and reinstalled Git..... I don't know what I'm doing wrong...

**Comment by C3n7ral051nt4g3ncy** [comment]
@KMH418 As I said above, you maybe have to use :
`pip install -r requirements.txt` instead of `pip3`

**Reply by KMH418** [reply]
I did that too... it gives the same info.
This is also among the text that comes back with that code above:
WARNING: Ignoring invalid distribution -ip (c:\users\chari\appdata\local\programs\python\python310\lib\site-packages)
Its listed several times in yellow. Is it relevant to my issue?

Crazy thing is: I just tried another repository and it works... so IDK...

**Reply by KMH418** [reply]
OK so I did a command with just pip and then with just pip3 to see what came back. It shows the commands that pip recognizes. Now, I'm not a coder so there is a 10,000% chance I'm wrong.... but in the commands that pip responds to on my system, there is no -r like there are with other letters. 
Since they both (pip install -r requirements.txt and pip3 install -r requirements.txt) both bring back the exact same info, I am sure my observation above is irrelevant.

**Reply by C3n7ral051nt4g3ncy** [reply]
@KMH418: 
https://note.nkmk.me/en/python-pip-install-requirements

**Reply by KMH418** [reply]
OK I might have found something..... In that link, there is a code: pip freeze

I ran that and it is showing the repository that I did get to work in the list of things it pulled up. It does not show the Socid...

**Reply by KMH418** [reply]
HOLY SHITE!!!! I got it to do something....!! 
cd socid-extractor
python setup.py install
socid_extractor --url https://www.deviantart.com/muse1908

That brought up:
C:\Users\chari\socid-extractor>socid_extractor --url https://www.deviantart.com/muse1908
Analyzing URL https://www.deviantart.com/muse1908...
country: France
created_at: 2005-06-16 14:17:41 UTC
gender: female
username: MuseMercier
website: www.purelymuse.com
links: ['https://www.instagram.com/muse.mercier/', 'https://twitter.com/musenews']
tagline: Nothing worth having is easy...
bio: Hi! My name is Muse Mercier, I'm an English artist living in France!\n \nI'm the subject in all my images here. My amazingly talented husband David helps me create the photographs. I'm a trained professional stage actress, but I took the leap to full-time online nude artist when the pandemic struck and the theatres closed!\nI'm also a genuine naturist, hence all the lovely natural nakedness! :sun: \n\nI fund my nude art (and pay my bills) purely via my Patreon, without it the majority of my work couldn't exist: \n\nhttps://www.patreon.com/musemercier  (1000s of images/videos)\n\nI hope you'll enjoy all the images I post for you! 🥰\n\n\nAll images copyright Muse1908 2021. No unauthorise d use.

**Reply by KMH418** [reply]
I went through what I did for the one that did work and applied it to this one...

**Reply by C3n7ral051nt4g3ncy** [reply]
That's great news! Now you have it.
All good, just adapt the commands to what you want to look for !

**Reply by C3n7ral051nt4g3ncy** [reply]
> I went through what I did for the one that did work and applied it to this one...

Nice. Glad I stay away from Windows though! It's a headache.

**Reply by KMH418** [reply]
Thank you SOOO Much! You are awesome! Sorry I took up so much of your Labor Day but hope you have a fantastic rest of it!! Thanks again!

**Reply by C3n7ral051nt4g3ncy** [reply]
No problem I was working online, it only takes a few seconds to reply 👍 
Glad you have it sorted.
Click on `answered` so the thread can be closed. 🔒
